### ISSUE TITLE
Monad Comprehension

### ISSUE THREAD

**Issue by Hjkiiiim** [nus-cs2030/2324-s2#809]
<img width="526" alt="Screenshot 2024-04-29 at 3 29 46 PM" src="https://github.com/nus-cs2030/2324-s2/assets/163497439/ccfdb3c9-9c49-421b-b03c-b9bdb0140dec">

Hi, could anyone explain how this do { ... } construct works and when should we use it? 
Thanks:)

**Comment by vanesssalai** [comment]
Hi the do{...} is pseudocode, so it doesn't do anything

**Comment by sc2603** [comment]
Yep it helps you conceptualize the steps needed to solve a problem without getting bogged down in syntax details.
